compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -339,7 +339,7 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
      *  or a TermRef to a singleton value. These are
      *  the base elements required to generate a mirror.
      */
-    def reduce(mirroredType: Type)(using Context): Either[String, MirrorSource] = mirroredType match
+    def reduce(mirroredType: Type)(using Context): Either[String, MirrorSource] = mirroredType.normalized match
```

Comment:
maybe we can explicitly do a check for AppliedType of a match type, (alongside the Tuple case), then normalise there

Reply:
Or maybe from `Implicits.inferImplicitArg` before even entering the Syntheziser. Is there any reason not to ?

Reply:
normalize is known to be a slow operation, so should be used only when necessary